persistent-test/src/DeleteCascadeTest.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+module DeleteCascadeTest where
+
+import Init
+import Database.Persist.TH (mkDeleteCascade)
+
+
+share [mkPersist sqlSettings, mkMigrate "migrateAll", mkDeleteCascade sqlSettings] [persistLowerCase|
+PersonX
+    username String
+    name String
+    age Int Maybe
+    UniqueName name
+    Primary username
+    deriving Show
+BlogPostX
+    title String
+    authorId PersonXId
+    deriving Show
+AliasX
+    person String
+    alias String
+    -- TODO: Uncomment when we can implement non-primary key foreign key
+    -- references.
+    -- Foreign Person fkperson person name
+    deriving Show
+|]
+
+specsWith :: (MonadIO m, MonadFail m) => RunDb SqlBackend m -> Spec
+specsWith runDb = describe "DeleteCascade" $ do
+    it "should delete by primary key reference" $ runDb $ do
+        pk <-insert (PersonX "username0" "name0" Nothing)
+        insertMany_ [BlogPostX "b0" pk, BlogPostX "b1" pk]
+        i <- length <$> selectList [BlogPostXAuthorId ==. pk] []
+        liftIO $ i `shouldBe` 2
+        deleteCascade pk
+        xs <- selectList [BlogPostXAuthorId ==. pk] []
+        liftIO $ void xs `shouldBe` []
+
+    it "should delete by non-primary key refernece" $ do
+        pendingWith "Depends on specifying non-PK references: see issue #909"
+        runDb $ do
+            pk <-insert (PersonX "username0" "name0" Nothing)
+            ak <- insert (AliasX "name0" "othername")
+            deleteCascade pk
+            ma <- get ak
+            liftIO $ void ma `shouldBe` Nothing
```

Comment:
@rvl This should take care of your test case, right?

Reply:
@parsonsmatt Thanks for looking at this and trying the fix. Yes it does match my use case. I can work around it by adding more `deleteWhere` statements though. If `persistent-sqlite` could add foreign key constraints to the table, that would be ideal, although I sense there might be some limitation of persistent and SQLite together that prevents this.

Reply:
I think that `persistent-sqlite` does add foreign key indexes. It doesn't specify the cascade mode though. Typically you'd write something like:

```sql
CREATE TABLE foo (
  blah_id INT REFERENCES blah(id) ON DELETE CASCADE
);
```

and when the relevant `blah` gets deleted, the `foo` entries get deleted as well.

I do want to extend the QQ syntax with a cascade-mode for deletes and updates. Something like:

```
Foo
   blahId  BlahId OnDelete Cascade OnUpdate Restrict
```

would be reflected in the generated SQL migrations and then you wouldn't need the "manual" cascading entries.